ArchiSteamFarm/GitHub.cs

```diff
@@ -0,0 +1,220 @@
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Markdig;
+using Markdig.Renderers;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ArchiSteamFarm {
+	internal static class GitHub {
+		private static MarkdownDocument ExtractChangelogFromBody(string markdownText) {
+			if (string.IsNullOrEmpty(markdownText)) {
+				ASF.ArchiLogger.LogNullError(nameof(markdownText));
+				return null;
+			}
+
+			MarkdownDocument markdownDocument = Markdown.Parse(markdownText);
+			bool insideChangelog = false;
+
+			foreach (Block block in markdownDocument.ToList()) {
+				if (!insideChangelog) {
+					if (block is HeadingBlock headingBlock && (headingBlock.Inline.FirstChild != null) && headingBlock.Inline.FirstChild is LiteralInline literalInline && (literalInline.Content.ToString() == "Changelog")) {
+						insideChangelog = true;
+					}
+
+					markdownDocument.Remove(block);
+					continue;
+				}
+
+				if (block is ThematicBreakBlock) {
+					insideChangelog = false;
+					markdownDocument.Remove(block);
+				}
+			}
+
+			return markdownDocument;
+		}
+
+		internal static async Task<List<ReleaseResponse>> GetReleases(byte count) {
+			if (count == 0) {
+				ASF.ArchiLogger.LogNullError(nameof(count));
+				return null;
+			}
+
+			string releaseURL = SharedInfo.GithubReleaseURL + "?per_page=" + count;
+
+			return await GetReleasesFromURL(releaseURL).ConfigureAwait(false);
+		}
+
+		internal static async Task<ReleaseResponse> GetLatestRelease(bool stable = true) {
+			string releaseURL = SharedInfo.GithubReleaseURL + (stable ? "/latest" : "?per_page=1");
+
+			if (stable) {
+				return await GetReleaseFromURL(releaseURL).ConfigureAwait(false);
+			} else {
+				return (await GetReleasesFromURL(releaseURL).ConfigureAwait(false))?.ElementAt(0);
+			}
+		}
+
+		internal static async Task<ReleaseResponse> GetRelease(string version) {
+			if (string.IsNullOrEmpty(version)) {
+				ASF.ArchiLogger.LogNullError(nameof(version));
+				return null;
+			}
+
+			return await GetReleaseFromURL(SharedInfo.GithubReleaseURL + "/tags/" + version).ConfigureAwait(false);
+		}
+
+		private static async Task<ReleaseResponse> GetReleaseFromURL(string releaseURL) {
+			if (string.IsNullOrEmpty(nameof(releaseURL))) {
+				ASF.ArchiLogger.LogNullError(nameof(releaseURL));
+				return null;
+			}
+
+			WebBrowser.ObjectResponse<ReleaseResponse> objectResponse = await Program.WebBrowser.UrlGetToJsonObject<ReleaseResponse>(releaseURL).ConfigureAwait(false);
+			if (objectResponse?.Content == null) {
```

Comment:
This is pointless since you're not doing anything anyway, just `return objectResponse?.Content`.